include/gsl/gsl

```diff
@@ -146,27 +142,23 @@ private:
 namespace std
 {
 template <class T>
-struct hash<gsl::not_null<T>>
+struct hash<::gsl::not_null<T>>
 {
```

Comment:
I understand where you're coming from, but please, let's not write it that way.  Several instances below and elsewhere.